Index: component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/helpers/CustomRequestWrapper.java
===================================================================
--- component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/helpers/CustomRequestWrapper.java	(revision 52253)
+++ component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/helpers/CustomRequestWrapper.java	(working copy)
@@ -175,8 +175,13 @@
 
   @Override
   public String getParameter(String name) {
-    return parameterMap.get(name) != null && parameterMap.get(name).length != 0 ? parameterMap.get(name)[0]
-                                                                               : null;
+    if (super.getParameter(Constants.COMPONENT_PARAMETER) != null
+        && super.getParameter(Constants.COMPONENT_PARAMETER).equals(this.windowId)) {
+      return parameterMap.get(name) != null && parameterMap.get(name).length != 0 ? parameterMap.get(name)[0]
+                                                                                 : super.getParameter(name);
+    } else {
+      return null;
+    }
   }
 
   @Override
