Index: component/core/src/main/java/org/exoplatform/services/portletcontainer/pci/model/Portlet.java
===================================================================
--- component/core/src/main/java/org/exoplatform/services/portletcontainer/pci/model/Portlet.java	(revision 64715)
+++ component/core/src/main/java/org/exoplatform/services/portletcontainer/pci/model/Portlet.java	(revision 64716)
@@ -208,12 +208,12 @@
   public final int getPortletSessionScope() {
     if (getContainerRuntimeOption() != null) {
       String[] valuesPortlet = getContainerRuntimeOption().get(
-          "javax.portlet.includedPortletSessionScope");
+          "javax.portlet.servletDefaultSessionScope");
       if ((valuesPortlet != null) && valuesPortlet[0].equals(PCConstants.PORTLET_SCOPE))
         return PortletSession.PORTLET_SCOPE;
     }
     String[] valuesApplication = application.getContainerRuntimeOption().get(
-        "javax.portlet.includedPortletSessionScope");
+        "javax.portlet.servletDefaultSessionScope");
     if ((valuesApplication != null) && valuesApplication[0].equals(PCConstants.PORTLET_SCOPE))
       return PortletSession.PORTLET_SCOPE;
 
