Index: component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/StateAwareResponseImp.java
===================================================================
--- component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/StateAwareResponseImp.java	(revision 65575)
+++ component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/portletAPIImp/StateAwareResponseImp.java	(working copy)
@@ -200,9 +200,8 @@
    * @return render parameters
    * @see javax.portlet.StateAwareResponse#getRenderParameterMap()
    */
-  public final Map getRenderParameterMap() {
-    // TODO
-    return null;
+  public final Map<java.lang.String,java.lang.String[]> getRenderParameterMap() {
+    return ((EventOutput) this.getOutput()).getRenderParameters();
   }
 
   /**
