Index: component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/aop/PortletCacheCommand.java
===================================================================
--- component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/aop/PortletCacheCommand.java	(revision 71574)
+++ component/plugins/pc/src/main/java/org/exoplatform/services/portletcontainer/plugins/pc/aop/PortletCacheCommand.java	(working copy)
@@ -157,8 +157,6 @@
           req.getInput().getInternalWindowID().getPortletName()).hashCode());
       return req.getInput().getInternalWindowID().getOwner() + uniqueId + modifier;
     }
-    if (req.getSession(false) == null)
-      return null;
     uniqueId = req.getInput().getInternalWindowID().getUniqueID();
     return req.getSession().getId() + req.getRemoteUser() + req.getInput().getInternalWindowID().getOwner() + uniqueId
       + modifier;
